lib/cc/engine/analyzers/python/node.rb

```diff
@@ -0,0 +1,74 @@
+module CC
+  module Engine
+    module Analyzers
+      module Python
+        class Node
+          SCRUB_PROPERTIES = ["_type", "attributes"].freeze
+
+          def initialize(node, file, default_line = 0)
+            @node = node
+            @file = file
+
+            set_default_line(default_line)
+          end
+
+          def format
+            if @node.is_a?(Hash)
+              type = @node["_type"].to_sym
+
+              if valid_properties
+                create_sexp(type, *properties_to_sexps)
+              else
+                type
+              end
+            elsif @node.is_a?(Array)
+              @node.map do |n|
+                Node.new(n, @file, @line).format
+              end
+            end
+          end
+
+          private
+
+          def properties_to_sexps
+            valid_properties.map do |key, value|
+              if value.is_a?(Array)
+                create_sexp(key.to_sym, *Node.new(value, @file, @line).format)
+              elsif value.is_a?(Hash)
+                create_sexp(key.to_sym, *Node.new(value, @file, @line).format)
```

Comment:
Since we're treating Arrays and Hashes the same way, could we combine the conditions?